File: CSharp/PaymentsService/Infrastructure/Payments/GooglePay/GooglePayPaymentFactory.cs
using PaymentsService.Domain.Payments;

namespace PaymentsService.Infrastructure.Payments.GooglePay;

public sealed class GooglePayPaymentFactory : IPaymentAbstractFactory
{
    public IPaymentClient GetPaymentClient()
    {
        return new GooglePayPaymentClient();
    }

    public IProcessPaymentRequestValidator GetProcessPaymentRequestValidator()
    {
        return GooglePayProcessPaymentRequestValidator.Instance;
    }

    public ProcessPaymentBaseResponse GetValidationFailedResponse(RequestValidationResult result)
    {
        return new ProcessGooglePayPaymentResponse()
        {
            Status = "ValidationFailed",
            Error = result.Error
        };
    }
}

File: Python/PaymentsService/Infrastructure/Payments/GooglePay/GooglePayPaymentFactory.py
from Domain.Payments.IPaymentAbstractFactory import IPaymentAbstractFactory
from Domain.Payments.IPaymentClient import IPaymentClient
from Domain.Payments.IProcessPaymentRequestValidator import IProcessPaymentRequestValidator
from Domain.Payments.ProcessPaymentBaseResponse import ProcessPaymentBaseResponse
from Infrastructure.Payments.GooglePay.GooglePayPaymentClient import GooglePayPaymentClient
from Infrastructure.Payments.GooglePay.GooglePayProcessPaymentRequestValidator import GooglePayProcessPaymentRequestValidator
from Infrastructure.Payments.GooglePay.ProcessGooglePayPaymentResponse import ProcessGooglePayPaymentResponse


class GooglePayPaymentFactory(IPaymentAbstractFactory):
    def get_payment_client(self) -> IPaymentClient:
        return GooglePayPaymentClient()

    def get_process_payment_request_validator(self) -> IProcessPaymentRequestValidator:
        return GooglePayProcessPaymentRequestValidator()

    def get_validation_failed_response(self, result) -> ProcessPaymentBaseResponse:
        return ProcessGooglePayPaymentResponse(status="ValidationFailed", error=result.error)